services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -710,8 +717,638 @@ func TestCreateApplicationVersion(t *testing.T) {
+					Version:       1,
+					LockBehaviour: api.LockBehavior_Fail,
+				},
+				&DeployApplicationVersion{
+					Environment:   envAcceptance,
+					Application:   "app3",
+					Version:       1,
+					LockBehaviour: api.LockBehavior_Fail,
+				},
+			},
+			ExistentCommitPaths: []string{
+				"commits/aa/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/applications/app1/.empty",
+				"commits/aa/bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb/applications/app2/.empty",
+				"commits/aa/cccccccccccccccccccccccccccccccccccccc/applications/app3/.empty",
+			},
+		},
+		{
+			Name: "Create several applications from the same SHA1 commit ID",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "acceptance",
+					Config:      config.EnvironmentConfig{Upstream: &config.EnvironmentConfigUpstream{Environment: envAcceptance, Latest: false}},
+				},
+				&CreateApplicationVersion{
+					Application:    "app1",
+					SourceCommitId: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+					Manifests: map[string]string{
+						envAcceptance: "acceptance",
+					},
+				},
+				&CreateApplicationVersion{
+					Application:    "app2",
+					SourceCommitId: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+					Manifests: map[string]string{
+						envAcceptance: "acceptance",
+					},
+				},
+				&CreateApplicationVersion{
+					Application:    "app3",
+					SourceCommitId: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+					Manifests: map[string]string{
+						envAcceptance: "acceptance",
+					},
+				},
+				&DeployApplicationVersion{
+					Environment:   envAcceptance,
+					Application:   "app1",
+					Version:       1,
+					LockBehaviour: api.LockBehavior_Fail,
+				},
+				&DeployApplicationVersion{
+					Environment:   envAcceptance,
+					Application:   "app2",
+					Version:       1,
+					LockBehaviour: api.LockBehavior_Fail,
+				},
+				&DeployApplicationVersion{
+					Environment:   envAcceptance,
+					Application:   "app3",
+					Version:       1,
+					LockBehaviour: api.LockBehavior_Fail,
+				},
+			},
+			ExistentCommitPaths: []string{
+				"commits/aa/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/applications/app1/.empty",
+				"commits/aa/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/applications/app2/.empty",
+				"commits/aa/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa/applications/app3/.empty",
+			},
+		},
+		{
+			Name: "Create application with non SHA1 commit ID",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "acceptance",
+					Config:      config.EnvironmentConfig{Upstream: &config.EnvironmentConfigUpstream{Environment: envAcceptance, Latest: false}},
+				},
+				&CreateApplicationVersion{
+					Application:    "app",
+					SourceCommitId: "nonsense",
+					Manifests: map[string]string{
+						envAcceptance: "acceptance",
+					},
+				},
+				&DeployApplicationVersion{
+					Environment:   envAcceptance,
+					Application:   "app",
+					Version:       1,
+					LockBehaviour: api.LockBehavior_Fail,
+				},
```

Comment:
Please add here "NonExistentCommitPaths" with the "nonsense" string from above, e.g. "no/nsense..."

Reply:
Good point, added now.